elements/pl-symbolic-input/pl-symbolic-input.py

```diff
@@ -29,11 +34,32 @@ def get_variables_list(variables_string):
     else:
         return []
 
+def prepare_context(element_html, data, problem_name):
+    if problem_name in phs.CONSTANTS_INSTANCE_LOOKUP.keys():
+        return    
+    phs.CONSTANTS_INSTANCE_LOOKUP[problem_name] = phs._Constants()
+    element = lxml.html.fragment_fromstring(element_html)
+    extra_functions_string = pl.get_string_attrib(element, OPTION_ENABLE_EXTRA_FUNCTION_NAMES, EXTRA_FUNCTIONS_DEFAULT)
+    extra_functions = get_variables_list(extra_functions_string)
+    phs.CONSTANTS_INSTANCE_LOOKUP[problem_name] = phs.CONSTANTS_INSTANCE_LOOKUP[problem_name].add_new_operator_names(extra_functions)
+    custom_functions_by_group_string = pl.get_string_attrib(element, OPTION_ENABLE_CUSTOM_FUNCTION_GROUPS, CUSTOM_FUNCTION_GROUPS_DEFAULT)
+    phs.CONSTANTS_INSTANCE_LOOKUP[problem_name] = phs.CONSTANTS_INSTANCE_LOOKUP[problem_name].enable_extended_func_names(custom_functions_by_group_string)
+    custom_constants_by_group_string = pl.get_string_attrib(element, OPTION_ENABLE_SYMBOLIC_CONSTANTS, SYMBOLIC_CONSTANT_GROUPS_DEFAULT)
+    phs.CONSTANTS_INSTANCE_LOOKUP[problem_name] = phs.CONSTANTS_INSTANCE_LOOKUP[problem_name].enable_extended_symbolic_constants(problem_name, custom_constants_by_group_string)
```

Comment:
@jonatanschroeder 
It is organized that way since the array of classes that ``phs.CONSTANTS_INSTANCE_LOOKUP`` has type defined in the ``phs`` Python file. Is this really a major stopping point? Tomatoes versus tomahhtos, right?

Reply:
It is good programming practice to store variables in the context in which they are used. There is no good reason to store it there. The fact that the type is defined in `phs` doesn't mean the variable/array should be stored there as well.